pytorch_lightning/trainer/trainer.py

```diff
@@ -1354,6 +1358,14 @@ def _run_sanity_check(self, ref_model):
             # restore the previous stage when the sanity check if finished
             self.state.stage = stage
 
+    def reset_train_dataloader(self, model: Optional["pl.LightningModule"] = None) -> None:
+        self._last_train_dl_reload_epoch = self.current_epoch
+        return super().reset_train_dataloader(model=model)
+
+    def reset_val_dataloader(self, model: Optional["pl.LightningModule"] = None) -> None:
+        self._last_val_dl_reload_epoch = self.current_epoch
+        return super().reset_val_dataloader(model=model)
```

Comment:
In the context of #10417, which proposes to move the data loading related methods and attributes to the data class, I think these changes should go to `data_loading.py`(where the reset method are).

Reply:
Sounds good! I moved the changes to `data_loading.py`